components/brave_wallet/browser/internal/hd_key_utils.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_UTILS_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_UTILS_H_
+
+#include <optional>
+#include <string_view>
+#include <vector>
+
+#include "base/containers/span.h"
+
+namespace brave_wallet {
+
+namespace internal {
+void SecureZeroBuffer(base::span<uint8_t> data);
+}
+
+inline constexpr char kMasterNode[] = "m";
+inline constexpr uint32_t kHardenedOffset = 0x80000000;
+
+// Parses BIP-32 full derivation path into a vector of indexes. Hardened indexes
+// expected to end with a single quote per BIP-44 style.
+// https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+// https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+// TODO(apaymyshev): We most likely do not need this parsing as at runtime all
+// parsing is done from hardcoded strings. Worth implementing account roots
+// derivations in terms of DeriveHardened/DeriveNormal calls.
+std::optional<std::vector<uint32_t>> ParseFullHDPath(std::string_view path);
+
+class ScopedSecureZeroSpan {
```

Comment:
the filename should be scoped_secure_zero_span to match the class name

Reply:
I guess there is also `ParseFullHDPath` above so this should go in a separate file

Reply:
actually we should be using `SecureAllocator` here

Reply:
actually this doesn't seem related to the change in implementation and has some other issues related to base::span storage